dev/bots/analyze-sample-code.dart

```diff
@@ -403,7 +413,9 @@ class SampleChecker {
         }
         if (!inSampleSection) {
           final Match sampleMatch = _dartDocSampleBeginRegex.firstMatch(trimmedLine);
-          if (line == '// Examples can assume:') {
+          if (line == '// Examples are not null safe.') {
+            samplesAreNullSafe = false;
+          } else if (line == '// Examples can assume:') {
```

Comment:
Instead of having a separate "magic comment", could we just declare it something like:
```dart
// Examples can assume:
// // @dart = 2.9
// var foo = bar;
```

The `@dart` comment shouldn't affect the whole file because it won't come before the imports in the host file. It would also give some control as to which version of NBD (as opposed to NNBD) is wanted (although I'm not sure that's needed).

Alternatively, we could make it be just that the `// Examples are not null safe.` comment must come inside (probably first line?) of the `// Examples can assume:` magic comment block.

I'm just loathe to introduce another magic comment.

Reply:
Changed it to your suggestion.